pkg/vm/vm_test.go

```diff
@@ -472,68 +443,24 @@ func TestPushm1to16(t *testing.T) {
 	}
 }
 
-func TestPushData1BadNoN(t *testing.T) {
-	prog := []byte{byte(opcode.PUSHDATA1)}
-	vm := load(prog)
-	checkVMFailed(t, vm)
-}
-
-func TestPushData1BadN(t *testing.T) {
-	prog := []byte{byte(opcode.PUSHDATA1), 1}
-	vm := load(prog)
-	checkVMFailed(t, vm)
-}
-
-func TestPushData1Good(t *testing.T) {
-	prog := makeProgram(opcode.PUSHDATA1, 3, 1, 2, 3)
-	vm := load(prog)
-	runVM(t, vm)
-	assert.Equal(t, 1, vm.estack.Len())
-	assert.Equal(t, []byte{1, 2, 3}, vm.estack.Pop().Bytes())
-}
-
-func TestPushData2BadNoN(t *testing.T) {
-	prog := []byte{byte(opcode.PUSHDATA2)}
-	vm := load(prog)
-	checkVMFailed(t, vm)
-}
-
-func TestPushData2ShortN(t *testing.T) {
-	prog := []byte{byte(opcode.PUSHDATA2), 0}
-	vm := load(prog)
-	checkVMFailed(t, vm)
-}
-
-func TestPushData2BadN(t *testing.T) {
-	prog := []byte{byte(opcode.PUSHDATA2), 1, 0}
-	vm := load(prog)
-	checkVMFailed(t, vm)
-}
-
-func TestPushData2Good(t *testing.T) {
-	prog := makeProgram(opcode.PUSHDATA2, 3, 0, 1, 2, 3)
-	vm := load(prog)
-	runVM(t, vm)
-	assert.Equal(t, 1, vm.estack.Len())
-	assert.Equal(t, []byte{1, 2, 3}, vm.estack.Pop().Bytes())
-}
-
-func TestPushData4BadNoN(t *testing.T) {
-	prog := []byte{byte(opcode.PUSHDATA4)}
-	vm := load(prog)
-	checkVMFailed(t, vm)
+func TestPUSHDATA1(t *testing.T) {
+	t.Run("Good", getTestFuncForVM([]byte{byte(opcode.PUSHDATA1), 3, 1, 2, 3}, []byte{1, 2, 3}))
+	t.Run("Good", getTestFuncForVM([]byte{byte(opcode.PUSHDATA1)}, nil))
```

Comment:
Why is it `Good`?

Reply:
Other than that, the refactoring part is actually very nice.

Reply:
Copy-paste, fixed.